dicomanonymizer/simpledicomanonymizer.py

```diff
@@ -249,8 +249,61 @@ def anonymizeDataset(dataset, extraAnonymizationRules = None):
     if extraAnonymizationRules != None:
         currentAnonymizationActions.update(extraAnonymizationRules)
 
+    from pydicom.tag import Tag
+
+    privateTags = []
+
     for tag, action in currentAnonymizationActions.items():
         action(dataset, tag)
+        try:
+            element = dataset.get(tag)
+        except:
+            print("Cannot get element from tag : ", tag)
+
+        if element and element.tag.is_private:
+            elementValue = element.value
+            tagGroup = element.tag.group
+            # The element is a private creator
+            if elementValue in dataset.private_creators(tagGroup):
+                creator = {
+                    "tagGroup": tagGroup,
+                    "creatorName": element.value
+                }
+                privateElement = None
+            # The element is a private element with an asociated private creator
+            else:
+                # Shift the element tag in order to get the creatorTag
+                # 0x1009 >> 8 will give 0x0010
+                creatorTagElement = element.tag.element >> 8
+                creatorTag = Tag(tagGroup, creatorTagElement)
+                creatorDataset = dataset.get(creatorTag)
+                creator = {
+                    "tagGroup": tagGroup,
+                    "creatorName": creatorDataset.value
+                }
+                # Define which offset should be applied to the creator to find
+                # this element
+                # 0x0010 << 8 will give 0x1000
+                offsetFromCreator = element.tag.element - (creatorTagElement << 8)
+                privateElement = {
+                    "element": element,
+                    "offset": offsetFromCreator
+                }
+
+            private = {
+                "creator": creator,
+                "element": privateElement
+            }
+
+            privateTags.append(private)
 
     # X - Private tags = (0xgggg, 0xeeee) where 0xgggg is odd
     dataset.remove_private_tags()
+
+    # Adding back private tags if specified by user
+    for privateTag in privateTags:
+        creator = privateTag["creator"]
+        element = privateTag["element"]
+        block = dataset.private_block(creator["tagGroup"], creator["creatorName"], create=True)
```

Comment:
move line inside the if (no need to call private_block if element is None)

Reply:
No we can't do this. If we have at least one private tag, then it's mandatory to have at least one creator.
A creator can leave alone but an element can't. It's possible to only keep a creator only.